cmake/legacy.cmake

```diff
@@ -295,6 +295,8 @@ if(PACKAGE_SET STREQUAL full)
     URL https://root.cern/download/root_v${root_version}.source.tar.gz
     URL_HASH SHA256=d02f224b4908c814a99648782b927c353d44db79dea2cadea86138c1afc23ae9
     ${CMAKE_DEFAULT_ARGS} CMAKE_ARGS
+      "-Dbuiltin_gsl=ON"
+      "-Dbuiltin_xrootd=ON"
```

Comment:
Before we use this solution I would like to try overwriting the upstream `xrootd` package. Will get back to you this week. In any way, the solution should be conditioned to only be effective on macOS, imho.